test/functional/collection.test.js

```diff
@@ -1307,4 +1312,34 @@ describe('Collection', function() {
       });
     }
   });
+
+  it('should createIndex with serverSelection readPreference primary', {
```

Comment:
Lookin pretty good, I have a few thoughts:
- Can we make this a data-driven test so we're testing all [DDL](https://www.geeksforgeeks.org/sql-ddl-dql-dml-dcl-tcl-commands/) operations? 
- Since testing all of those operation types is likely to be hairy (`create` for database and collection, for instance), I think it makes sense to develop a more comprehensive mock for `Topology`. I'm thinking something along the lines of:
```
it('should do something with the mock topology', function(done) {
   const client = new MongoClient(...);
   const topology = mockTopology(client); // <-- `client` being passed in anticipates a future where `client.topology` is set during construction, for now you would have to stub the `Topology` class. 
   client.connect(err => {
      client.db('test').collection('foo').createIndex({ quote: 'text' }, err => {
         expect(err).to.not.exist;
         expect(topology).property('operations[0]').to.equal({
           /*
             `operations` would conform to some shape that we think gives us all the info we need for assertions, maybe:
             `{ readPreference: ..., command: ... }` to start
           */
         });

         client.close(done);
      });
   });
});
```

I think we can do this work as a follow-on PR, maybe another approach is better but I think we definitely want to ensure we have tests for all commands which we expect to explicitly force a primary `readPreference`